code/client/src/pages/Create.jsx

```diff
@@ -60,52 +76,72 @@ const Create = () => {
   const [progressStage, setProgressStage] = useState(0)
   const [address, setAddress] = useState() // '0x12345678901234567890'
   const [effectiveTime, setEffectiveTime] = useState()
+  const [doubleOtp, setDoubleOtp] = useState(false)
 
   const [durationVisible, setDurationVisible] = useState(false)
-  const [section, setSection] = useState(2)
+  const [section, setSection] = useState(sectionViews.setupOtp)
   const [qrCodeData, setQRCodeData] = useState()
   const [otp, setOtp] = useState('')
 
   const [deploying, setDeploying] = useState()
 
   const otpRef = useRef()
 
-  const getQRCodeUri = () => {
+  const getQRCodeUri = (otpSeed) => {
     // otpauth://TYPE/LABEL?PARAMETERS
-    return `otpauth://totp/${name}?secret=${b32.encode(seed)}&issuer=Harmony`
+    return `otpauth://totp/${name}?secret=${b32.encode(otpSeed)}&issuer=Harmony`
   }
+
   useEffect(() => {
     (async function () {
-      const uri = getQRCodeUri()
+      const otpSeed = section === sectionViews.setupSecondOtp ? seed2 : seed
+
+      const uri = getQRCodeUri(otpSeed)
+
       const data = await qrcode.toDataURL(uri, { errorCorrectionLevel: 'low', width: isMobile ? 192 : 256 })
+
       setQRCodeData(data)
     })()
   }, [name])
 
   useEffect(() => {
-    if (section === 2 && worker) {
+    if (section === sectionViews.setupOtp && worker) {
       console.log('posting to worker')
       const t = Math.floor(Date.now() / WalletConstants.interval) * WalletConstants.interval
       setEffectiveTime(t)
       worker && worker.postMessage({
-        seed, effectiveTime: t, duration, slotSize, interval: WalletConstants.interval
+        seed, seed2, effectiveTime: t, duration, slotSize, interval: WalletConstants.interval
       })
     }
   }, [section, worker])
 
   useEffect(() => {
+    const settingUpSecondOtp = section === sectionViews.setupSecondOtp
+
     if (otp.length !== 6) {
       return
     }
-    const expected = ONEUtil.genOTP({ seed })
+
+    const currentSeed = settingUpSecondOtp ? seed2 : seed
+
+    const expected = ONEUtil.genOTP({ seed: currentSeed })
+
     const code = new DataView(expected.buffer).getUint32(0, false).toString()
+
+    setOtp('')
+
     if (code.padStart(6, '0') !== otp.padStart(6, '0')) {
-      console.log(`Expected: ${code}. Got: ${otp}`)
       message.error('Code is incorrect. Please try again.')
-      setOtp('')
+
+      otpRef?.current?.focusInput(0)
+    } else if (doubleOtp && !settingUpSecondOtp) {
+      setSection(sectionViews.setupSecondOtp)
+
+      setName(`${name} (2nd)`)
```

Comment:
It would be incorrect to change the name state variable because the wallet would be saved as `${name} (2nd)` in redux

Reply:
Good point. So we still want to save the name as original `name` in the state, but the authenticator would have a different name for the second OTP (that is only append (2nd) text).